test/e2e/framework/ingress_utils.go

```diff
@@ -1109,14 +1139,50 @@ func (j *IngressTestJig) GetRootCA(secretName string) (rootCA []byte) {
 
 // TryDeleteIngress attempts to delete the ingress resource and logs errors if they occur.
 func (j *IngressTestJig) TryDeleteIngress() {
-	err := j.Client.ExtensionsV1beta1().Ingresses(j.Ingress.Namespace).Delete(j.Ingress.Name, nil)
+	if err := j.RunDelete(j.Ingress, j.Class); err != nil {
+		Logf("Error while deleting the ingress %v/%v with class %s: %v", j.Ingress.Namespace, j.Ingress.Name, j.Class, err)
+	}
+}
+
+// RunDelete runs the required command to delete the given ingress.
+func (j *IngressTestJig) RunDelete(ing *extensions.Ingress, class string) error {
```

Comment:
Name it `DeleteIngress`? BTW the second argument `class` is unused :)

Reply:
same as above